operator-common/src/test/java/io/strimzi/operator/PlatformFeaturesAvailabilityTest.java

```diff
@@ -7,157 +7,155 @@
-                "  \"compiler\": \"gc\",\n" +
-                "  \"platform\": \"linux/amd64\"\n" +
-                "}";
-
-        HttpServer mockHttp = startMockApi(context, version, Collections.EMPTY_LIST);
-
-        KubernetesClient client = new DefaultKubernetesClient("127.0.0.1:" + mockHttp.actualPort());
-
-        Checkpoint a = context.checkpoint();
-
-        PlatformFeaturesAvailability.create(vertx, client).onComplete(context.succeeding(pfa -> context.verify(() -> {
-            assertThat("Versions are not equal", pfa.getKubernetesVersion(), is(KubernetesVersion.V1_20));
-            stopMockApi(context, mockHttp);
-            a.flag();
-        })));
+    @Disabled
+    void testFalsePositiveCanary(VertxTestContext context) {
+        verify(context, pfa -> assertThat("Answer to the Ultimate Question of Life, the Universe, and Everything", is(42)));
     }
 
     @Test
-    public void testVersionDetectionMinikube(VertxTestContext context) throws InterruptedException {
-        String version = "{\n" +
-                "  \"major\": \"1\",\n" +
-                "  \"minor\": \"20\",\n" +
-                "  \"gitVersion\": \"v1.20.1\",\n" +
-                "  \"gitCommit\": \"c4d752765b3bbac2237bf87cf0b1c2e307844666\",\n" +
-                "  \"gitTreeState\": \"clean\",\n" +
-                "  \"buildDate\": \"2020-12-18T12:09:25Z\",\n" +
-                "  \"goVersion\": \"go1.15.5\",\n" +
-                "  \"compiler\": \"gc\",\n" +
-                "  \"platform\": \"linux/amd64\"\n" +
-                "}";
-
-        HttpServer mockHttp = startMockApi(context, version, Collections.EMPTY_LIST);
-
-        KubernetesClient client = new DefaultKubernetesClient("127.0.0.1:" + mockHttp.actualPort());
-
-        Checkpoint async = context.checkpoint();
-
-        PlatformFeaturesAvailability.create(vertx, client).onComplete(context.succeeding(pfa -> context.verify(() -> {
-            assertThat("Versions are not equal", pfa.getKubernetesVersion(), is(KubernetesVersion.V1_20));
-            stopMockApi(context, mockHttp);
-            async.flag();
-        })));
+    public void testVersionDetectionOpenShift(VertxTestContext context) {
+        version = () ->  "{\n" +
+                        "  \"major\": \"1\",\n" +
+                        "  \"minor\": \"20\",\n" +
+                        "  \"gitVersion\": \"v1.20.1\",\n" +
+                        "  \"gitCommit\": \"c4d752765b3bbac2237bf87cf0b1c2e307844666\",\n" +
+                        "  \"gitTreeState\": \"clean\",\n" +
+                        "  \"buildDate\": \"2020-12-18T12:00:47Z\",\n" +
+                        "  \"goVersion\": \"go1.15.5\",\n" +
+                        "  \"compiler\": \"gc\",\n" +
+                        "  \"platform\": \"linux/amd64\"\n" +
+                        "}";
+
+        verify(context, pfa -> assertThat("Versions are not equal", pfa.getKubernetesVersion(), is(KubernetesVersion.V1_20)));
```

Comment:
How does this create the `PlatformFeaturesAvailability`?

Reply:
```
    private void verify(VertxTestContext context, Consumer<PlatformFeaturesAvailability> assertions) {
        Handler<PlatformFeaturesAvailability> runAssertions = pfa -> context.verify(() -> assertions.accept(pfa));

        PlatformFeaturesAvailability.create(vertx, kubernetesClient)
                                    .onComplete(context.succeeding(runAssertions))
                                    .onComplete(unused -> context.completeNow());
    }
```

But! I'll be revising this to be a lot less clever in PRv2.

Reply:
Same as mentioned above with the `version`. I think you should name this better to make the code more readable. If you would call it `verifyPlatformFeaturesAvailability(...)` it would be already much  easier to understand.